microcosm_eventsource/stores.py

```diff
@@ -54,6 +54,7 @@ def upsert_on_index_elements(self, instance):
 
         """
         with self.flushing():
+            instance.id = self.new_object_id()
```

Comment:
Previously new events with a parent would not use the Store's mock friendly `new_object_id()` function.

Reply:
I think this works because we do an `ON CONFLICT DO NOTHING` (vs trying to update an existing instance with a new `id`), but my preference would be to  inject the id at a higher level of abstraction; this function takes a fully instantiated `instance` and doesn't definitively create anything.

I wonder if we can inject the object id before the `instance` is passed in to this function?